Added Washington Post Collection, Document, and Experiments

Kytabyte opened the pull request.

Peilin-Yang commented on src/main/java/io/anserini/document/WaPostDocument.java:
```diff
@@ -0,0 +1,98 @@
+                    try {
+                        builder.append(removeTags(contentObj.getString(PARAGRAPH_TAG).trim())).append("\n");
+                    } catch (JSONException e) {
+                        continue;
```
Is this critical? For common practice we throw error to interrupt the indexing. 
Otherwise it is hard to tell if it is a bug in the code or a corrupt document.

Peilin-Yang commented on src/main/java/io/anserini/document/WaPostDocument.java:
```diff
@@ -0,0 +1,98 @@
+                    }
+                }
+            } else {
+                // LOG.warn("No type or content tag defined in Article " + id + ", ignored...");
```
Add something meaningful? e.g. uncomment the LOG.

lintool commented on src/main/java/io/anserini/collection/WaPostCollection.java:
```diff
@@ -0,0 +1,57 @@
+
+package io.anserini.collection;
+
+import io.anserini.document.WaPostDocument;
```
Rename from `WaPost` to `WashPost`.

lintool commented on docs/experiments-core17.md:
```diff
@@ -19,12 +19,31 @@ index segment into one.
 For the retrieval model: specify `-bm25` to use BM25, `-ql` to use query likelihood, and add `-rm3` to invoke the RM3 
 relevance feedback model (requires docvectors index).
 
+
```
Reformat this to follow the other collections? I.e., make the documentation look consistent.

Kytabyte replied:
I observed that many of the experiment docs have this line with the same wording

lintool commented on docs/experiments-wapo.md:
````diff
@@ -0,0 +1,31 @@
+
+Indexing:
+
+```bash
````
Make docs look consistent with other collections.

lintool commented on src/main/java/io/anserini/document/WaPostDocument.java:
```diff
@@ -0,0 +1,98 @@
+/**
+ * A Washington Post document.
+ */
+public class WaPostDocument implements SourceDocument {
```
`WaPost` -> `WashPost`

lintool commented on src/main/resources/topics-and-qrels/topics.core17.nist.txt:
```diff
@@ -0,0 +1,598 @@
+<top>
```
rename to `topics.core17.txt` and `qrels.core17.txt`

lintool commented:
I was able to get NYT indexed on `tuna`, per #217. Did you get it working?

lintool commented:
Additional comments in #217

Kytabyte commented:
@Peilin-Yang I've added test cases and rewritten the catch exceptions.
@lintool Renamed the files and updated format

You can ignore the tf-idf dump utils as I moved it to another branch and it will no longer be involved in this pull request

Kytabyte commented on docs/experiments-core17.md:
```diff
@@ -22,9 +22,29 @@ After indexing is done, you should be able to perform a retrieval as follows:
+
+If everything goes correctly, you should be able to replicate the following results:
+
+| Metric | BM25   | QL     | RM3    |
```
Right, this is RM3 based on QL. I'll do the other now.

Peilin-Yang commented:
code lgtm however I am lack of context in terms of the initiative and the accuracy of the results.
